api/backend/ecs/service_manager.go

```diff
@@ -120,12 +120,28 @@ func (this *ECSServiceManager) DeleteService(environmentID, serviceID string) er
 	ecsServiceID := id.L0ServiceID(serviceID).ECSServiceID()
 	desiredCount := int64(0)
 
-	if err := this.ECS.UpdateService(ecsEnvironmentID.String(), ecsServiceID.String(), nil, &desiredCount); err != nil {
-		return err
+	service, err := this.GetService(environmentID, serviceID)
+        if err != nil {
+                return  err
+        }
+
+        taskARNs := []*string{}
+        for _, deployment := range service.Deployments {
+                arns, err := getTaskARNs(this.ECS, ecsEnvironmentID, stringp(deployment.DeploymentID))
+                if err != nil {
+                        return err
+                }
+
+                taskARNs = append(taskARNs, arns...)
+        }
+
+	for _, arn := range taskARNs{
+		if err := this.ECS.StopTask(ecsEnvironmentID.String(), "Service deleted by User", *arn); err != nil{
+			log.Warnf("Stop Task for Service '%s' had error: %v", serviceID, err)
+		}
 	}
 
-	log.Debugf("Waiting for service to stop")
-	if err := this.waitUntilServiceStopped(ecsEnvironmentID, ecsServiceID); err != nil {
```

Comment:
can you think of an event where forcefully stopping a service would be undesirable in comparison to the current behavior? i.e. the difference between killing a process and waiting for it to gracefully shutdown versus `kill -9`? I suppose this necessitates us having a better understanding of how `StopTask` works...